pep/app/services/current-user.ts

```diff
@@ -216,11 +215,11 @@ export default class CurrentUserService extends Service {
      * @param {Document} document
      * @memberof CurrentUserService
      */
-    addDocument(key: PreferenceDocumentsKey, document: Document) {
+    addDocument(key: PreferenceDocumentsKey, documentId: string) {
```

Comment:
at the very least i think these methods should be renamed, e.g. `addPrefDocument()` but I still dont think they really belong in the current-user service at all, especially since updating these can now be a one-liner.

`this.loadLocalStoragePrefs();` should not be necessary since the `currentUser.preferences` should always be up-to-date with what's in localStorage. All this logic could be condensed and done directly in the search item component as:
```ts
this.currentUser.updatePrefs({
    [PreferencesKey.FAVORITES]: [...this.currentUser.preferences[PreferencesKey.FAVORITES], id].uniq()
})
```

Reply:
I find it cleaner to abstract that out

i.e. 
`this.currentUser.addDocument()`
is a lot cleaner than 
```
this.currentUser.updatePrefs({
    [PreferencesKey.FAVORITES]: [...this.currentUser.preferences[PreferencesKey.FAVORITES], id].uniq()
})
```
and since we add and remove from multiple places I think the way I have it provides a DRY'er experience

Reply:
But I can change the name